internal/ocm/resource/cluster.go

```diff
@@ -125,11 +144,13 @@ func (c *Cluster) CreateAWSBuilder(awsTags map[string]string, ec2MetadataHttpTok
 		awsBuilder.Tags(awsTags)
 	}
 
-	ec2MetadataHttpTokensVal := cmv1.Ec2MetadataHttpTokensOptional
-	if ec2MetadataHttpTokens != nil {
-		ec2MetadataHttpTokensVal = cmv1.Ec2MetadataHttpTokens(*ec2MetadataHttpTokens)
+	if clusterTopology == rosa.Classic {
+		ec2MetadataHttpTokensVal := cmv1.Ec2MetadataHttpTokensOptional
+		if ec2MetadataHttpTokens != nil {
+			ec2MetadataHttpTokensVal = cmv1.Ec2MetadataHttpTokens(*ec2MetadataHttpTokens)
+		}
+		awsBuilder.Ec2MetadataHttpTokens(ec2MetadataHttpTokensVal)
 	}
```

Comment:
```suggestion
if clusterTopology == rosa.Classic {
		if ec2MetadataHttpTokens != nil {
			awsBuilder.Ec2MetadataHttpTokens(cmv1.Ec2MetadataHttpTokens(*ec2MetadataHttpTokens))
		} else {
			awsBuilder.Ec2MetadataHttpTokens(cmv1.Ec2MetadataHttpTokensOptional)
		}
	}
```

Reply:
I guess this is more of a preference, I find picking the default first and not having the else clearer :laughing:

Reply:
I can also just go for the builder directly with the Optional default and remove the variable anyways, so that would be better and we still don't have the else. Thanks for shedding some light :+1: